src/tools/terraform.ts

```diff
@@ -0,0 +1,97 @@
+import {Tool} from './tool'
+import * as fs from 'fs'
+import * as path from 'path'
+import {ChildProcess, spawn, SpawnOptions} from 'child_process'
+import {GeneralError} from '../error'
+import {LoggerFactory} from '../util/logger'
+import Axios from 'axios'
+
+// Terraform install settings
+const {bin, dest, url} = require('./terraform/settings.ts')
+
+export class Terraform extends Tool {
+  static logger = LoggerFactory.createLogger(Terraform)
+
+  // Installer
+  async install(): Promise<string> {
```

Comment:
install should return the folder where the executable is or the path to the executable